learning/courses/quantum-chem-with-vqe/geometry.ipynb

```diff
@@ -0,0 +1,610 @@
+    "\n",
+    "First, we load some of the packages we used before and define the cholesky function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0a5d39bd-8c26-404f-8057-c29e3af70df4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qiskit.quantum_info import SparsePauliOp\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "#!pip install pyscf==2.4.0\n",
+    "from pyscf import ao2mo, gto, mcscf, scf\n",
+    "\n",
+    "\n",
+    "def cholesky(V, eps):\n",
+    "    # see https://arxiv.org/pdf/1711.02242.pdf section B2\n",
+    "    # see https://arxiv.org/abs/1808.02625\n",
+    "    # see https://arxiv.org/abs/2104.08957\n",
+    "    no = V.shape[0]\n",
+    "    chmax, ng = 20 * no, 0\n",
+    "    W = V.reshape(no**2, no**2)\n",
+    "    L = np.zeros((no**2, chmax))\n",
+    "    Dmax = np.diagonal(W).copy()\n",
+    "    nu_max = np.argmax(Dmax)\n",
+    "    vmax = Dmax[nu_max]\n",
+    "    while vmax > eps:\n",
+    "        L[:, ng] = W[:, nu_max]\n",
+    "        if ng > 0:\n",
+    "            L[:, ng] -= np.dot(L[:, 0:ng], (L.T)[0:ng, nu_max])\n",
+    "        L[:, ng] /= np.sqrt(vmax)\n",
+    "        Dmax[: no**2] -= L[: no**2, ng] ** 2\n",
+    "        ng += 1\n",
+    "        nu_max = np.argmax(Dmax)\n",
+    "        vmax = Dmax[nu_max]\n",
+    "    L = L[:, :ng].reshape((no, no, ng))\n",
+    "    print(\n",
+    "        \"accuracy of Cholesky decomposition \",\n",
+    "        np.abs(np.einsum(\"prg,qsg->prqs\", L, L) - V).max(),\n",
+    "    )\n",
+    "    return L, ng\n",
+    "\n",
+    "\n",
+    "def identity(n):\n",
+    "    return SparsePauliOp.from_list([(\"I\" * n, 1)])\n",
+    "\n",
+    "\n",
+    "def creators_destructors(n, mapping=\"jordan_wigner\"):\n",
+    "    c_list = []\n",
+    "    if mapping == \"jordan_wigner\":\n",
+    "        for p in range(n):\n",
+    "            if p == 0:\n",
+    "                ell, r = \"I\" * (n - 1), \"\"\n",
+    "            elif p == n - 1:\n",
+    "                ell, r = \"\", \"Z\" * (n - 1)\n",
+    "            else:\n",
+    "                ell, r = \"I\" * (n - p - 1), \"Z\" * p\n",
+    "            cp = SparsePauliOp.from_list([(ell + \"X\" + r, 0.5), (ell + \"Y\" + r, -0.5j)])\n",
+    "            c_list.append(cp)\n",
+    "    else:\n",
+    "        raise ValueError(\"Unsupported mapping.\")\n",
+    "    d_list = [cp.adjoint() for cp in c_list]\n",
+    "    return c_list, d_list"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "40e25b54-7927-4dbb-a26f-1c6b33f7f349",
+   "metadata": {},
+   "source": [
+    "Now to define our Hamiltonian, we will use PySCF exactly as in the previous example, but now we will include a variable \"x\" to play the role of our interatomic distance. This will return the core energy, single-electron energy, and two-electron energies as before."
```

Comment:
```suggestion
    "Now to define our Hamiltonian, we will use PySCF exactly as in the previous example, but now we will include a variable, `x`, to play the role of our interatomic distance. This will return the core energy, single-electron energy, and two-electron energies as before."
```